### PR TITLE
[AUTO][IE TESTS] Fix IE test issue for auto-device plugin

### PR DESCRIPTION
**Pull request by coneypo**
### Details:
 - Will not skip test cases for auto-device plugin, `ASSERT_THROW` exception when using incorrectConfig;

### Tickets:
 - *57552*

### CODE REVIEW

**Comment by coneypo** [comment]
Currenlty only `PERF_COUNT` and `AUTO_DEVICE_LIST` are accepted by auto plugin;

File: inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp [hunk 1]
```diff
@@ -72,11 +72,15 @@ namespace BehaviorTestsDefinitions {
...
+        if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos) {
+            auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration);
         } else {
-            ASSERT_THROW(auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration),
-                         InferenceEngine::Exception);
+            // "PERF_COUNT" is accepted by auto-device plugin
+            if (configuration.find("PERF_COUNT") == configuration.end()) {
+                ASSERT_THROW(auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration),
+                             InferenceEngine::Exception);
+            } else {
```
Comment (myshevts) [inline inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp:82]: else?

File: inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp [hunk 2]
```diff
@@ -72,11 +72,15 @@ namespace BehaviorTestsDefinitions {
...
         InferenceEngine::CNNNetwork cnnNet(function);
-        if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) != std::string::npos) {
-            GTEST_SKIP();
+        if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos) {
+            auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration);
         } else {
-            ASSERT_THROW(auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration),
-                         InferenceEngine::Exception);
+            // "PERF_COUNT" is accepted by auto-device plugin
+            if (configuration.find("PERF_COUNT") == configuration.end()) {
```
Comment (myshevts) [inline inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp:79]: end() points to the past-the-end element in the map :) 
also being in the end != that conf is ONLY PERF_COUNT

File: inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp [hunk 3]
```diff
@@ -143,17 +148,17 @@ namespace BehaviorTestsDefinitions {
             targetDevice.find(CommonTestUtils::DEVICE_HETERO) == std::string::npos) {
             ASSERT_NO_THROW(ie->SetConfig(config, targetDevice));
         }
-        // Load CNNNetwork to target plugins
         if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos) {
             auto execNet = ie->LoadNetwork(cnnNet, targetDevice, config);
             execNet.CreateInferRequest();
+        } else {
```
Comment (myshevts) [inline inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp:154]: pls adda comment why the AUTO should throw

**Review by myshevts — CHANGES_REQUESTED** [review]

**Comment by myshevts** [comment]
pls add a test entry (to the config, that the AUto DOESN'T THROW for "AUTO_",
this was not added to the https://github.com/openvinotoolkit/openvino/pull/5979 from whta I see

**Comment by coneypo** [comment]
LoadNetwork with incorrect or not supported config for auto-device plugin will throw in `check_config`, so remove config test for smoke_Auto_BehaviorTests;

**Comment by coneypo** [comment]
> pls add a test entry (to the config, that the AUto DOESN'T THROW for "AUTO_",
> this was not added to the #5979 from whta I see

`CorrectAutoConfigTests` added:

- CorrectAutoConfigTests/SetCorrectConfig
- CorrectAutoConfigTests/CanLoadNetworkWithCorrectConfig

File: inference-engine/include/ie_plugin_config.hpp [hunk 4]
```diff
@@ -296,6 +296,11 @@ DECLARE_CONFIG_KEY(DYN_BATCH_LIMIT);
  */
 DECLARE_CONFIG_KEY(DYN_BATCH_ENABLED);
 
+/**
+ * @brief Reserved configuration for Auto-Device plugin.
+ */
+DECLARE_CONFIG_KEY(AUTO_);
```
Comment (mashoujiang) [inline inference-engine/include/ie_plugin_config.hpp:302]: why adding this config key? if you want to test, please add to test file config, not here

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp [hunk 5]
```diff
@@ -42,7 +42,14 @@ namespace {
     };
 
     const std::vector<std::map<std::string, std::string>> AutoConfigs = {
-            {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_CPU}}
+        {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_CPU},
+            {InferenceEngine::PluginConfigParams::KEY_PERF_COUNT, "YES"}},
+        {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_AUTO},
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:47]: this is incorrect, AUTO:AUTO is not allowed

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp [hunk 6]
```diff
@@ -83,7 +90,9 @@ namespace {
 
     const std::vector<std::map<std::string, std::string>> autoinconfigs = {
         {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_CPU},
-            {InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, "OFF"}}
+            {InferenceEngine::PluginConfigParams::KEY_PERF_COUNT, "10"}},
+        {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_AUTO},
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:94]: same as above

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/config.cpp [hunk 7]
```diff
@@ -104,75 +101,74 @@ namespace {
...
 
     const std::vector<std::map<std::string, std::string>> autoconf = {
-            {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_GPU}}
-    };
-
-    const std::vector<std::map<std::string, std::string>> auto_cpu_gpu_conf = {
-        {{InferenceEngine::KEY_AUTO_DEVICE_LIST , std::string(CommonTestUtils::DEVICE_CPU) + "," + CommonTestUtils::DEVICE_GPU}}
+            {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_GPU},
+                    {InferenceEngine::PluginConfigParams::KEY_PERF_COUNT, "YES"}},
+            {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_AUTO},
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/config.cpp:106]: dont use AUTO:AUTO, it should throw exception. but if you want to test AUTO:AUTO use case  with expected behavior, that's ok.

File: inference-engine/include/ie_plugin_config.hpp (hunk 4 above)
Comment (myshevts) [reply inference-engine/include/ie_plugin_config.hpp]: good catch @mashoujiang ! I personally  have no much time to review secondary PRs like this, so pls take the full responsibility.

Comment (mashoujiang) [reply inference-engine/include/ie_plugin_config.hpp]: sure, no problem :)

**Comment by coneypo** [comment]
@mashoujiang CI is green now, please review this PR:)

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp [hunk 8]
```diff
@@ -12,11 +12,18 @@ namespace {
...
-    const std::vector<std::map<std::string, std::string>> conf = {
+    const std::vector<std::map<std::string, std::string>> config = {
             {}
     };
 
-    const std::vector<std::map<std::string, std::string>> Configs = {
+    INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests, CorrectConfigAPITests,
+                             ::testing::Combine(
+                                     ::testing::ValuesIn(netPrecisions),
+                                     ::testing::Values(CommonTestUtils::DEVICE_CPU),
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:22]: why do we need add CPU test case in this PR?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp [hunk 9]
```diff
@@ -26,7 +33,25 @@ namespace {
             {{InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_LIMIT, "10"}}
     };
 
-    const std::vector<std::map<std::string, std::string>> MultiConfigs = {
+    INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests, CorrectConfigTests,
+                             ::testing::Combine(
+                                     ::testing::ValuesIn(netPrecisions),
+                                     ::testing::Values(CommonTestUtils::DEVICE_CPU),
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:39]: same as above, CPU?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp [hunk 10]
```diff
@@ -26,7 +33,25 @@ namespace {
...
+                             CorrectConfigTests::getTestCaseName);
+
+    const std::vector<std::map<std::string, std::string>> correctMultiConfig = {
+            {{InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES , CommonTestUtils::DEVICE_CPU}}
+    };
+
+    INSTANTIATE_TEST_SUITE_P(smoke_Multi_BehaviorTests, CorrectConfigAPITests,
+                             ::testing::Combine(
+                                     ::testing::ValuesIn(netPrecisions),
+                                     ::testing::Values(CommonTestUtils::DEVICE_MULTI),
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:50]: MULTI?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp [hunk 11]
```diff
@@ -41,38 +66,48 @@ namespace {
...
             {{InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, "OFF"}},
             {{InferenceEngine::PluginConfigParams::KEY_CPU_BIND_THREAD, "OFF"}},
             {{InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_LIMIT, "NAN"}}
     };
 
-    const std::vector<std::map<std::string, std::string>> multiinconfigs = {
+    INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests, IncorrectConfigTests,
+                             ::testing::Combine(
+                                     ::testing::ValuesIn(netPrecisions),
+                                     ::testing::Values(CommonTestUtils::DEVICE_CPU),
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:99]: again as above

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/infer_request.cpp [hunk 12]
```diff
@@ -14,39 +14,39 @@ namespace {
...
-
     INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests, InferRequestTests,
                             ::testing::Combine(
                                     ::testing::ValuesIn(netPrecisions),
                                     ::testing::Values(CommonTestUtils::DEVICE_CPU),
-                                    ::testing::ValuesIn(configs)),
+                                    ::testing::ValuesIn(correctConfigs)),
                             InferRequestTests::getTestCaseName);
 
+    const std::vector<std::map<std::string, std::string>> incorrectMultiConfigs = {
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/infer_request.cpp:30]: is this incorrect multi config tests?

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/config.cpp [hunk 13]
```diff
@@ -13,71 +13,19 @@ namespace {
...
-                ::testing::Values(CommonTestUtils::DEVICE_GPU),
-                ::testing::ValuesIn(inconfigs)),
-            IncorrectConfigTests::getTestCaseName);
-
-    INSTANTIATE_TEST_SUITE_P(smoke_Multi_BehaviorTests, IncorrectConfigTests,
-            ::testing::Combine(
-                ::testing::ValuesIn(netPrecisions),
-                ::testing::Values(CommonTestUtils::DEVICE_MULTI),
-                ::testing::ValuesIn(multiinconfigs)),
-            IncorrectConfigTests::getTestCaseName);
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/config.cpp:65]: why delete these MULTI and GPU test cases?

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/config.cpp [hunk 14]
```diff
@@ -99,80 +47,117 @@ namespace {
...
 
-    INSTANTIATE_TEST_SUITE_P(smoke_AutoCG_BehaviorTests, IncorrectConfigAPITests,
+    INSTANTIATE_TEST_SUITE_P(smoke_BehaviorTests, IncorrectConfigAPITests,
                             ::testing::Combine(
-                                ::testing::ValuesIn(netPrecisions),
-                                ::testing::Values(CommonTestUtils::DEVICE_AUTO),
-                                ::testing::ValuesIn(autoinconfigs)),
+                                    ::testing::ValuesIn(netPrecisions),
+                                    ::testing::Values(CommonTestUtils::DEVICE_GPU),
+                                    ::testing::ValuesIn(config)),
```
Comment (mashoujiang) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/behavior/config.cpp:135]: IncorrectConfig tests better use inconfig, also I think it is not necessary to change the original tests.

**Review by mashoujiang — COMMENTED** [review]
Too many unnecessary code modification (code format, code line movement), which makes code review unclear. if this PR is just for AUTO plugin tests cases, better not change other tests (unless they have explict errors).

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp (hunk 8 above)
Comment (coneypo) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:22]: not added, just modified code location, will revert these changes

**Review by mashoujiang — APPROVED** [review]
@myshevts Looks OK now

File: inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp [hunk 15]
```diff
@@ -42,10 +42,8 @@ TEST_P(InferConfigTests, canSetExclusiveAsyncRequests) {
         ASSERT_NO_THROW(ie->SetConfig(config, targetDevice));
     }
     // Load CNNNetwork to target plugins
-    if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos) {
-        auto execNet = ie->LoadNetwork(cnnNet, targetDevice, config);
-        execNet.CreateInferRequest();
-    }
+    auto execNet = ie->LoadNetwork(cnnNet, targetDevice, config);
+    execNet.CreateInferRequest();
```
Comment (myshevts) [inline inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp:46]: BTW, although not related directly to this PR, but I see no diff in the code of this test and the test below (withoutExclusiveAsyncRequests),
I think the **below** test should set the EXCLUSIVE_ASYNC_REQUESTS to NO (as it name implies).
@apankratovantonp and @ilya-lavrenov fyi

**Review by myshevts — APPROVED** [review]

File: inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp (hunk 15 above)
Comment (myshevts) [reply inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp:46]: @ilya-lavrenov pls comment, so @coneypo can fix the test if it is really incorrect (as his PR touches these tests)

Comment (ilya-lavrenov) [reply inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp:46]: @myshevts I agree, should be fixed

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp (hunk 15 above)
Comment (myshevts) [reply inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp:46]: I suggest to change the second (duplicated) test to set EXCLUSIVE_ASYNC_REQUESTS to NO and remove the checks on the #executors.

**Pull request merged**